src/NzbDrone.Core/Notifications/Webhook/WebhookSettings.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using FluentValidation;
+using NzbDrone.Core.Annotations;
+using NzbDrone.Core.ThingiProvider;
+using NzbDrone.Core.Validation;
+using NzbDrone.Core.Validation.Paths;
+
+namespace NzbDrone.Core.Notifications.Webhook
+{
+    public class WebhookSettingsValidator : AbstractValidator<WebhookSettings>
+    {
+        public WebhookSettingsValidator()
+        {
+            RuleFor(c => c.Url).IsValidUrl();
+        }
+    }
+
+    public class WebhookSettings : IProviderConfig
+    {
+        private static readonly WebhookSettingsValidator Validator = new WebhookSettingsValidator();
+
+        [FieldDefinition(0, Label = "URL", Type = FieldType.Url)]
+        public String Url { get; set; }
```

Comment:
Okay I've already shutdown for the night so I'll push a new patch in the
morning with all the stylistic changes in the morning.

If I can I'll throw in the text only on grab too but that might not be till
evening.
On Aug 23, 2015 11:09 PM, "Mark McDowall" <notifications@github.com> wrote:

> In src/NzbDrone.Core/Notifications/Webhook/WebhookSettings.cs
> <https://github.com/Sonarr/Sonarr/pull/759#discussion_r37721558>:
>
> > +namespace NzbDrone.Core.Notifications.Webhook
> > +{
> > +    public class WebhookSettingsValidator : AbstractValidator<WebhookSettings>
> > +    {
> > +        public WebhookSettingsValidator()
> > +        {
> > +            RuleFor(c => c.Url).IsValidUrl();
> > +        }
> > +    }
> > +
> > +    public class WebhookSettings : IProviderConfig
> > +    {
> > +        private static readonly WebhookSettingsValidator Validator = new WebhookSettingsValidator();
> > +
> > +        [FieldDefinition(0, Label = "URL", Type = FieldType.Url)]
> > +        public String Url { get; set; }
>
> nit: string instead of String (things are a bit all over with this one,
> but we've landed on using string).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Sonarr/Sonarr/pull/759/files#r37721558>.
>